### PR TITLE
Address fragment accessors generation

### PR DESCRIPTION
**Pull request by sav007**
- Generate fragment accessor functions for named fragment interfaces
- Change the accessor naming: for inline fragment use `as*` prefix for named fragment accessor use just name of the fragment.
- Generate accessor functions as part of companion object

### CODE REVIEW

File: apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/backend/codegen/FragmentType.kt [hunk 1]
```diff
@@ -71,12 +83,5 @@ internal fun CodeGenerationAst.FragmentType.implementationTypeSpec(generateAsInt
       .toBuilder()
       .addSuperinterface(GraphqlFragment::class)
       .applyIf(generateAsInternal) { addModifiers(KModifier.INTERNAL) }
-      .addType(TypeSpec.companionObjectBuilder()
-          .addProperty(PropertySpec.builder("FRAGMENT_DEFINITION", String::class)
```
Comment (martinbonnin) [inline apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/backend/codegen/FragmentType.kt:75]: 👍 I guess we don't need that ?

File: apollo-compiler/src/test/graphql/com/example/fragment_in_fragment/AllStarship.kt [hunk 2]
```diff
@@ -233,8 +229,33 @@ class AllStarship : Query<AllStarship.Data, Operation.Variables> {
...
                       }
+
+                      companion object {
+                        fun Homeworld.planetFragment(): PlanetFragment? = this as? PlanetFragment
+
+                        fun Homeworld.pilotFragmentHomeworldPlanet():
+                            PilotFragment.Homeworld.Planet? = this as?
+                            PilotFragment.Homeworld.Planet
+
+                        fun Homeworld.starshipFragmentPilotConnectionEdgesNodePersonHomeworldPlanet():
```
Comment (martinbonnin) [inline apollo-compiler/src/test/graphql/com/example/fragment_in_fragment/AllStarship.kt:240]: That sounds like a little bit too much 🙃 . Feels like this shouldn't be generated

File: apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/backend/codegen/FragmentType.kt (hunk 1 above)
Comment (sav007) [reply apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/backend/codegen/FragmentType.kt:75]: Yep, for named fragment default implementation we don't need graphql definition as it already declared in interface

File: apollo-compiler/src/test/graphql/com/example/fragment_in_fragment/AllStarship.kt (hunk 2 above)
Comment (sav007) [reply apollo-compiler/src/test/graphql/com/example/fragment_in_fragment/AllStarship.kt:240]: Yeah, I guess we need to skip generating accessors for nested named fragments. Only accessors for top level fragments.

**Review by martinbonnin — APPROVED** [review]

**Pull request merged**